janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/function/mutate/CQLMutateManyUnloggedFunction.java

```diff
@@ -58,21 +63,37 @@ public void mutateMany(final Map<String, Map<StaticBuffer, KCVMutation>> mutatio
 
         final DistributedStoreManager.MaskedTimestamp commitTime = createMaskedTimestampFunction.apply(txh);
 
-        Optional<Throwable> errorAfterExecution = mutate(commitTime, mutations, txh);
+        List<CompletableFuture<AsyncResultSet>> resultList = new ArrayList<>();
 
-        if (errorAfterExecution.isPresent()) {
-            throw EXCEPTION_MAPPER.apply(errorAfterExecution.get());
+        mutations.forEach((tableName, tableMutations) -> {
+            final CQLKeyColumnValueStore columnValueStore = getColumnValueStore(tableName);
+
+            tableMutations.forEach((key, keyMutations) ->
+                toGroupedBatchableStatementsSequenceIterator(commitTime, keyMutations, columnValueStore, key)
+                    .forEach(group -> resultList.add(execAsyncUnlogged(group, txh))));
+        });
+
+        try {
+            CompletableFutureUtil.awaitAll(resultList);
+        } catch (Throwable e) {
+            throw EXCEPTION_MAPPER.apply(e);
         }
 
         sleepAfterWriteFunction.accept(commitTime);
     }
 
     protected CompletableFuture<AsyncResultSet> execAsyncUnlogged(Seq<BatchableStatement<BoundStatement>> group, StoreTransaction txh){
-        return this.session.executeAsync(
-            BatchStatement.newInstance(DefaultBatchType.UNLOGGED)
-                .addAll(group)
-                .setConsistencyLevel(getTransaction(txh).getWriteConsistencyLevel())
-        ).toCompletableFuture();
+        queryBackPressure.acquireBeforeQuery();
+        try{
+            return this.session.executeAsync(
+                BatchStatement.newInstance(DefaultBatchType.UNLOGGED)
+                    .addAll(group)
+                    .setConsistencyLevel(getTransaction(txh).getWriteConsistencyLevel())
+            ).whenComplete((asyncResultSet, throwable) -> queryBackPressure.releaseAfterQuery()).toCompletableFuture();
```

Comment:
Nice. I remember I did exact same thing when I was working with @cdegroc and saw immediate throughput improvement.